screen2.0/src/app/search/page.tsx

```diff
@@ -52,60 +85,129 @@ export default async function Search({
    */
   //This needs better input handling
   //Fails in basically any case when input isn't exactly as expected
-  const generateRows = (QueryResult: ApolloQueryResult<any>) => {
+  const generateRows = (QueryResult: ApolloQueryResult<any>,biosample: string = null) => {
     const rows: {
       //atac will need to be changed from string to number when that data is available
       accession: string
       class: string
       chromosome: string
       start: string
       end: string
-      dnase: number
+      dnase?: number
       atac: string
-      h3k4me3: number
-      h3k27ac: number
-      ctcf: number
+      h3k4me3?: number
+      h3k27ac?: number
+      ctcf?: number
     }[] = []
     const cCRE_data: cCREData[] = QueryResult.data.cCRESCREENSearch
+    
     let offset = 0
     cCRE_data.forEach((currentElement, index) => {
-      if (passesCriteria(currentElement)) {
+      if (passesCriteria(currentElement, biosample)) {
         rows[index - offset] = {
           accession: currentElement.info.accession,
           class: currentElement.pct,
           chromosome: currentElement.chrom,
           start: currentElement.start.toLocaleString("en-US"),
           end: (currentElement.start + currentElement.len).toLocaleString("en-US"),
-          dnase: +currentElement.dnase_zscore.toFixed(2),
+          dnase: biosample ? currentElement.ctspecific.dnase_zscore : currentElement.dnase_zscore,
           //Need to get this data still from somewhere
           atac: "TBD",
-          h3k4me3: +currentElement.promoter_zscore.toFixed(2),
-          h3k27ac: +currentElement.enhancer_zscore.toFixed(2),
-          ctcf: +currentElement.ctcf_zscore.toFixed(2),
+          h3k4me3: biosample ? currentElement.ctspecific.h3k4me3_zscore : currentElement.promoter_zscore,
+          h3k27ac: biosample ? currentElement.ctspecific.h3k27ac_zscore : currentElement.enhancer_zscore,
+          ctcf: biosample ? currentElement.ctspecific.ctcf_zscore : currentElement.ctcf_zscore,
         }
       }
       // Offset incremented to account for missing rows which do not meet filter criteria
       else {
         offset += 1
       }
     })
+    
     return rows
   }
 
-  const passesCriteria = (currentElement: cCREData) => {
-    //Chromatin Signals
-    if (
-      mainQueryParams.dnase_s < currentElement.dnase_zscore &&
-      currentElement.dnase_zscore < mainQueryParams.dnase_e &&
-      mainQueryParams.h3k4me3_s < currentElement.promoter_zscore &&
-      currentElement.promoter_zscore < mainQueryParams.h3k4me3_e &&
-      mainQueryParams.h3k27ac_s < currentElement.enhancer_zscore &&
-      currentElement.enhancer_zscore < mainQueryParams.h3k27ac_e &&
-      mainQueryParams.ctcf_s < currentElement.ctcf_zscore &&
-      currentElement.ctcf_zscore < mainQueryParams.ctcf_e
-    ) {
+
+  function passesCriteria(currentElement: cCREData,biosample: string = null) {
```

Comment:
filter criteria functions can be moved another file filters.ts

Reply:
Good idea, will move